collectors/VMPropertiesCollector.py

```diff
@@ -42,6 +42,7 @@ def do_metrics(self, target, g, i):
         if not token:
             print("skipping", target, "in", self.name, ", no token")
         uuids = self.target_vms[target]
+        project_ids = Resources.get_project_ids(target, token, uuids)
```

Comment:
This part needs to be moved. For every iteration of do_metrics (as in every thread) this is being freshly fetched. If it would be fetched in collect() and passed to do_metrics() would only run once and not on a per metric base.

Reply:
Ok, thanks for your comments on that. The first thing I solved in the collector to not die at any line. The second is a bit tricky, because I need the newest vm_uuids at that point to ensure an up to date status of vms in projects. I guess the newest match between project ids and vms are crucial. Let me think about that some more.

Reply:
You have to bear in mind, project ids aren't changing for VMs. This is a pretty static settings. Once a VM is deployed, it is hosted in the same project for its lifecycle in all.

Reply:
Thanks, that helps. With this information I can move things around.